torch/testing/_asserts.py

```diff
@@ -577,31 +577,43 @@ def _to_tensor(array_or_scalar_like: Any) -> Tuple[Optional[_TestingErrorMeta],
     return None, tensor
 
 
-def _to_tensor_pair(actual: Any, expected: Any) -> Tuple[Optional[_TestingErrorMeta], Optional[_TensorPair]]:
+def _check_types(actual: Any, expected: Any, *, allow_subclasses: bool) -> Optional[_TestingErrorMeta]:
+    # We exclude numbers here, since numbers of different type, e.g. int vs. float, should be treated the same as
+    # tensors with different dtypes. Without user input, passing numbers of different types will still fail, but this
+    # can be disabled by setting `check_dtype=False`.
+    if isinstance(actual, numbers.Number) and isinstance(expected, numbers.Number):
+        return None
+
+    msg_fmtstr = f"Except for Python scalars, {{}}, but got {type(actual)} and {type(expected)} instead."
+    common_superclass = isinstance(actual, type(expected)) or isinstance(expected, type(actual))
+    if not common_superclass:
+        return _TestingErrorMeta(AssertionError, msg_fmtstr.format("inputs are required to share a common superclass"))
+
+    if allow_subclasses or type(actual) is type(expected):
+        return None
+
+    return _TestingErrorMeta(AssertionError, msg_fmtstr.format("type equality is required"))
+
+
+def _to_tensor_pair(
+    actual: Any, expected: Any, *, allow_subclasses: bool
+) -> Tuple[Optional[_TestingErrorMeta], Optional[_TensorPair]]:
     """Converts a scalar-or-array-like pair to a :class:`_TensorPair`.
 
     Args:
         actual (Any): Actual array-or-scalar-like.
         expected (Any): Expected array-or-scalar-like.
+        allow_subclasses (bool): If ``True`` inputs of a common superclass are allowed. Otherwise type equality is
```

Comment:
I think this doc needs to be revised because, again, any two Python objects have a common superclass (object).

Reply:
Wait, shouldn't this say:

"If this is \`\`False\`\` (the default) then both inputs must have the same type. If \`\`True\`\` then one input may be a (direct or indirect) subclass of the other."

?